cds_ils/migrator/eitems/api.py

```diff
@@ -25,20 +25,32 @@
 from cds_ils.migrator.documents.api import get_all_documents_with_files, \
     get_documents_with_ebl_eitems, get_documents_with_external_eitems, \
     get_documents_with_proxy_eitems
-from cds_ils.migrator.errors import EItemMigrationError
+from cds_ils.migrator.errors import EItemMigrationError, FileMigrationError
+
+migrated_logger = logging.getLogger("migrated_records")
+records_logger = logging.getLogger("records_errored")
 
 
-def import_legacy_files(file_link):
+def import_legacy_files(file_link, file_checksum):
     """Download file from legacy."""
+
     # needed to ignore the migrator in the legacy statistics
     download_request_headers = {"User-Agent": "CDS-ILS Migrator"}
 
-    return io.BytesIO(
-        requests.get(
-            file_link, stream=True, headers=download_request_headers
-        ).content
+    file_response = requests.get(
+        file_link, stream=True, headers=download_request_headers
     )
 
+    imported_checksum = hashlib.md5(file_response.content).hexdigest()
```

Comment:
Instead of doing this, I would check after the creation of the file in CDS-ILS, e.g. (pseudocode):
```
ov = ObjectVersion.create(....)
assert file_checksum == ov.file_id.checksum
```

Reply:
done!